pkg/blockbuilder/types/grpc_transport.go

```diff
@@ -0,0 +1,147 @@
+package types
+
+import (
+	"context"
+	"flag"
+	"io"
+
+	"github.com/grafana/dskit/grpcclient"
+	"github.com/grafana/dskit/instrument"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/health/grpc_health_v1"
+
+	"github.com/grafana/loki/v3/pkg/blockbuilder/types/proto"
+	"github.com/grafana/loki/v3/pkg/util/constants"
+)
+
+var _ Transport = &GRPCTransport{}
+
+type GRPCTransportConfig struct {
```

Comment:
nit: should we embed this in builder config? ignore me if it is planned for a follow-up

Reply:
Yeah, I planned to leave this for when we're ready to wire it together.